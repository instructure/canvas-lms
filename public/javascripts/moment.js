define(['bower/moment/min/moment-with-locales'], function(moment) {
  // Include locale that is not a part of momentjs proper.
  var mi_nz = moment.defineLocale('mi-nz', {
    months: 'Kohi-tāte_Hui-tanguru_Poutū-te-rangi_Paenga-whāwhā_Haratua_Pipiri_Hōngoingoi_Here-turi-kōkā_Mahuru_Whiringa-ā-nuku_Whiringa-ā-rangi_Hakihea'.split('_'),
    monthsShort: 'Kohi_Hui_Pou_Pae_Hara_Pipi_Hōngoi_Here_Mahu_Whi-nu_Whi-ra_Haki'.split('_'),
    weekdays: 'Rātapu_Mane_Tūrei_Wenerei_Tāite_Paraire_Hātarei'.split('_'),
    weekdaysShort: 'Ta_Ma_Tū_We_Tāi_Pa_Hā'.split('_'),
    weekdaysMin: 'Ta_Ma_Tū_We_Tāi_Pa_Hā'.split('_'),
    longDateFormat: {
      LT: 'HH:mm',
      LTS: 'HH:mm:ss',
      L: 'DD/MM/YYYY',
      LL: 'D MMMM YYYY',
      LLL: 'D MMMM YYYY [i] HH:mm',
      LLLL: 'dddd, D MMMM YYYY [i] HH:mm'
    },
    calendar: {
      sameDay: '[i teie mahana, i] LT',
      nextDay: '[apopo i] LT',
      nextWeek: 'dddd [i] LT',
      lastDay: '[inanahi i] LT',
      lastWeek: 'dddd [whakamutunga i] LT',
      sameElse: 'L'
    },
    relativeTime: {
      future: 'i roto i %s',
      past: '%s i mua',
      s: 'te hēkona ruarua',
      m: 'he meneti',
      mm: '%d meneti',
      h: 'te haora',
      hh: '%d haora',
      d: 'he ra',
      dd: '%d ra',
      M: 'he marama',
      MM: '%d marama',
      y: 'he tau',
      yy: '%d tau'
    },
    ordinalParse: /\d{1,2}º/,
    ordinal: '%dº'
  });

  return moment;
});
