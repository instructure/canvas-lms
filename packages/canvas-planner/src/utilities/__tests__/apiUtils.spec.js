/*
 * Copyright (C) 2017 - present Instructure, Inc.
 *
 * This file is part of Canvas.
 *
 * Canvas is free software: you can redistribute it and/or modify it under
 * the terms of the GNU Affero General Public License as published by the Free
 * Software Foundation, version 3 of the License.
 *
 * Canvas is distributed in the hope that they will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
 * A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
 * details.
 *
 * You should have received a copy of the GNU Affero General Public License along
 * with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import moment from 'moment-timezone';
import {
  transformApiToInternalItem,
  transformInternalToApiItem,
  transformInternalToApiOverride,
  transformPlannerNoteApiToInternalItem
} from '../apiUtils';

const courses = [{
  id: '1',
  shortName: 'blah',
  image: 'blah_url',
  color: '#abffaa',
  informStudentsOfOverdueSubmissions: true,
}];
const groups = [{
  id: '9',
  assetString: 'group_9',
  name: 'group9',
  color: '#ffeeee',
  url: '/groups/9'
}];

function makeApiResponse (overrides = {}, assignmentOverrides = {}) {
  return {
    plannable_id: "10",
    context_type: "Course",
    course_id: "1",
    type: "submitting",
    ignore: `/api/v1/users/self/todo/assignment_10/submitting?permanent=0`,
    ignore_permanently: `/api/v1/users/self/todo/assignment_10/submitting?permanent=1`,
    visible_in_planner: true,
    planner_override: null,
    html_url: `/courses/1/assignments/10#submit`,
    plannable_type: 'assignment',
    plannable: makeAssignment(assignmentOverrides),
    submissions: false,
    new_activity: false,
    ...overrides,
  };
}

function makePlannerNoteApiResponse (overrides = {}) {
  return {
    id: 14,
    todo_date: '2017-06-21T18:58:51Z',
    title: 'abc123',
    details: 'asdfasdfasdf',
    user_id: 5,
    course_id: null,
    workflow_state: 'active',
    created_at: '2017-06-21T18:58:57Z',
    updated_at: '2017-06-21T18:58:57Z',
    ...overrides
  };
}

function makePlannerNote (overrides = {}) {
  return {
    id: 10,
    todo_date: "2017-05-19T05:59:59Z",
    title: "Some To Do Note",
    details: "Some To Do Note Details :)",
    user_id: '1',
    course_id: '1',
    workflow_state: 'active',
    ...overrides
  };
}

function makeAssignment (overrides = {}) {
  return {
    id: "10",
    description: "<p>Lorem ipsum etc.</p>",
    due_at: "2017-05-19T05:59:59Z",
    unlock_at: null,
    lock_at: null,
    points_possible: 100,
    grading_type: "points",
    assignment_group_id: "2",
    grading_standard_id: null,
    created_at: "2017-05-15T14:36:03Z",
    updated_at: "2017-05-15T16:20:35Z",
    peer_reviews: false,
    automatic_peer_reviews: false,
    position: 2,
    grade_group_students_individually: false,
    anonymous_peer_reviews: false,
    group_category_id: null,
    post_to_sis: false,
    moderated_grading: false,
    omit_from_final_grade: false,
    intra_group_peer_reviews: false,
    secure_params: "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9",
    course_id: "1",
    name: "",
    submission_types: [
      "online_text_entry",
      "online_upload"
    ],
    has_submitted_submissions: false,
    due_date_required: false,
    max_name_length: 255,
    in_closed_grading_period: false,
    is_quiz_assignment: false,
    muted: false,
    html_url: `/courses/1/assignments/10`,
    quiz_id: "1",
    anonymous_submissions: false,
    published: true,
    only_visible_to_overrides: false,
    locked_for_user: false,
    submissions_download_url: `/courses/1/quizzes/1/submissions?zip=1`,
    ...overrides,
  };
}

function makeDiscussionTopic (overrides = {}) {
  return {
    id: "1",
    title: "",
    last_reply_at: "2017-05-15T16:32:34Z",
    delayed_post_at: null,
    posted_at: "2017-05-15T16:32:34Z",
    assignment_id: 9,
    root_topic_id: null,
    position: null,
    podcast_has_student_posts: false,
    discussion_type: "side_comment",
    lock_at: null,
    allow_rating: false,
    only_graders_can_rate: false,
    sort_by_rating: false,
    user_name: "clay@instructure.com",
    discussion_subentry_count: 0,
    permissions: {
      attach: false,
      update: false,
      reply: true,
      delete: false
    },
    require_initial_post: null,
    user_can_see_posts: true,
    podcast_url: null,
    read_state: "unread",
    unread_count: 0,
    subscribed: false,
    topic_children: [],
    attachments: [],
    published: true,
    can_unpublish: false,
    locked: false,
    can_lock: false,
    comments_disabled: false,
    author: {
      id: "1",
      display_name: "Carl Chudyk",
      avatar_image_url: "http://canvas.instructure.com/images/messages/avatar-50.png",
      html_url: `/courses/1/users/1`
    },
    html_url: `/courses/1/discussion_topics/10`,
    url: `/courses/1/discussion_topics/10`,
    pinned: false,
    group_category_id: null,
    can_group: true,
    locked_for_user: false,
    message: "<p>Some prompt</p>",
    ...overrides,
  };
}

function makeGradedDiscussionTopic (overrides = {}) {
  return {
    id: "1",
    title: "",
    last_reply_at: "2017-05-15T16:32:34Z",
    delayed_post_at: null,
    posted_at: null,
    assignment_id: 10,
    root_topic_id: null,
    assignment: {due_at: "2017-05-15T16:32:34Z"},
    position: null,
    podcast_has_student_posts: false,
    discussion_type: "side_comment",
    lock_at: null,
    allow_rating: false,
    only_graders_can_rate: false,
    sort_by_rating: false,
    user_name: "clay@instructure.com",
    discussion_subentry_count: 0,
    permissions: {
      attach: false,
      update: false,
      reply: true,
      delete: false
    },
    require_initial_post: null,
    user_can_see_posts: true,
    podcast_url: null,
    read_state: "unread",
    unread_count: 0,
    subscribed: false,
    topic_children: [],
    attachments: [],
    published: true,
    can_unpublish: false,
    locked: false,
    can_lock: false,
    comments_disabled: false,
    author: {
      id: "1",
      display_name: "Carl Chudyk",
      avatar_image_url: "http://canvas.instructure.com/images/messages/avatar-50.png",
      html_url: `/courses/1/users/1`
    },
    html_url: `/courses/1/discussion_topics/10`,
    url: `/courses/1/discussion_topics/10`,
    pinned: false,
    group_category_id: null,
    can_group: true,
    locked_for_user: false,
    message: "<p>Some prompt</p>",
    ...overrides,
  };
}

function makeWikiPage(overrides = {}) {
  return {
    title: 'wiki_page title',
    created_at: "2017-06-16 10:08:00Z",
    url: 'wiki-page-title',
    editing_roles: 'teachers',
    page_id: '1',
    last_edited_by: {
      id: '1',
      display_name: 'Carl Chudyk',
      avatar_image_url: "http://canvas.instructure.com/images/messages/avatar-50.png",
      html_url: `/courses/1/users/1`,
    },
    published: true,
    hide_from_students: false,
    front_page: false,
    html_url: `/courses/1/pages/wiki-page-title`,
    todo_date: "2017-06-16 10:08:00Z",
    updated_at: "2017-06-16 10:08:00Z",
    locked_for_user: false,
    body: 'body of wiki page',
    ...overrides,
  };
}

describe('transformApiToInternalItem', () => {
  it('extracts and transforms the proper data for responses containing a status', () => {
    const apiResponse = makeApiResponse({
      submissions: {
        graded: true,
        has_feedback: true
      }
    });

    const result = transformApiToInternalItem(apiResponse, courses, groups, 'UTC');

    expect(result.status).toEqual({
      graded: true,
      has_feedback: true
    });
  });

  it('extracts and transforms the proper data for a quiz response', () => {
    const apiResponse = makeApiResponse({
      plannable_type: 'quiz',
      plannable: makeAssignment({
        name: 'How to make friends',
        submission_types: [ 'online_quiz' ],
      })
    });
    const result = transformApiToInternalItem(apiResponse, courses, groups, 'UTC');
    expect(result).toMatchSnapshot();
  });

  it('extracts and transforms the proper data for a graded discussion response', () => {
    const apiResponse = makeApiResponse({
      plannable_type: 'discussion_topic',
      plannable: makeGradedDiscussionTopic({
        name: 'How to make friends part 2',
        submission_types: [ 'discussion_topic' ],
      })
    });
    const result = transformApiToInternalItem(apiResponse, courses, groups, 'UTC');
    expect(result).toMatchSnapshot();
  });

  it('extracts and transforms the proper data for a discussion response', () => {
    const apiResponse = makeApiResponse({
      plannable_type: 'discussion_topic',
      plannable: makeDiscussionTopic({
        title: "How to make enemies",
        points_possible: 40,
        todo_date: "2017-05-19T05:59:59Z",
      })
    });
    const result = transformApiToInternalItem(apiResponse, courses, groups, 'UTC');
    expect(result).toMatchSnapshot();
  });

  it('extracts and transforms the proper data for a graded discussion reponse with an unread count', () => {
    const apiResponse = makeApiResponse({
      plannable_type: 'discussion_topic',
      submissions: { submitted: true },
      plannable: makeDiscussionTopic({
        title: "How to make enemies",
        points_possible: 40,
        todo_date: "2017-05-19T05:59:59Z",
        unread_count: 10
      })
    });
    const result = transformApiToInternalItem(apiResponse, courses, groups, 'UTC');
    expect(result).toMatchSnapshot();
  });

  it('extracts and transforms the proper data for an ungraded discussion reponse with an unread count', () => {
    const apiResponse = makeApiResponse({
      plannable_type: 'discussion_topic',
      submissions: false,
      plannable: makeDiscussionTopic({
        title: "How to make enemies",
        todo_date: "2017-05-19T05:59:59Z",
        unread_count: 10
      })
    });
    const result = transformApiToInternalItem(apiResponse, courses, groups, 'UTC');
    expect(result).toMatchSnapshot();
  });

  it('extracts and transforms the proper data for a assignment response', () => {
    const apiResponse = makeApiResponse({
      plannable_type: 'assignment',
      plannable: makeAssignment({
        points_possible: 50,
        name: "How to be neutral",
      }),
    });
    const result = transformApiToInternalItem(apiResponse, courses, groups, 'UTC');
    expect(result).toMatchSnapshot();
  });

  it('extracts and transforms the proper data for a planner_note response', () => {
    const apiResponse = makeApiResponse({
      context_type: undefined,
      course_id: undefined,
      plannable_type: 'planner_note',
      plannable: makePlannerNote()
    });

    const result = transformApiToInternalItem(apiResponse, courses, groups, 'UTC');
    expect(result).toMatchSnapshot();
  });

  it('extracts and transforms the proper data for a planner_note response without an associated course', () => {
    const apiResponse = makeApiResponse({
      context_type: undefined,
      course_id: undefined,
      plannable_type: 'planner_note',
      plannable: makePlannerNote({
        course_id: undefined
      })
    });

    const result = transformApiToInternalItem(apiResponse, courses, groups, 'UTC');
    expect(result).toMatchSnapshot();
  });

  it('extracts and transforms the proper data for an announcement response', () => {
    const apiResponse = makeApiResponse({
      plannable_type: 'announcement',
      plannable: makeDiscussionTopic({ // TODO: Discussion topic is probably fine for now to simulate this, but probably should change later
        due_at: undefined,
        todo_date: undefined,
        unread_count: 1
      })
    });

    const result = transformApiToInternalItem(apiResponse, courses, groups, 'UTC');
    expect(result).toMatchSnapshot();
  });

  it('extracts and transforms the ID for a wiki page repsonse', () => {
    const apiResponse = makeApiResponse({
      plannable_type: 'wiki_page',
      plannable: makeWikiPage({}),
    });
    const result = transformApiToInternalItem(apiResponse, courses, groups, 'UTC');
    expect(result.id).toEqual('1');
  });

  it('adds the dateBucketMoment field', () => {
    const apiResponse = makeApiResponse({
      plannable_type: 'assignment',
      plannable: makeAssignment({
        due_at: moment.tz('2017-05-24', 'Asia/Tokyo'),
      })
    });
    const result = transformApiToInternalItem(apiResponse, courses, groups, 'Europe/Paris');
    const expectedBucket = moment.tz('2017-05-23', 'Europe/Paris');
    expect(result.dateBucketMoment.isSame(expectedBucket)).toBeTruthy();
  });

  it('handles items without context (notes to self)', () => {
    const apiResponse = makeApiResponse();
    delete apiResponse.context;
    delete apiResponse.context_type;
    delete apiResponse.course_id;
    const result = transformApiToInternalItem(apiResponse, courses, groups, 'Europe/Paris');
    expect(result).toMatchObject({id: '10'});
  });

  it('throws if the timezone parameter is missing', () => {
    expect(() => transformApiToInternalItem({}, [])).toThrow();
  });

  it('copes with a non-existent (e.g. concluded) course', () => {
    const apiResponse = makeApiResponse({
      course_id: 999,
      plannable_type: 'planner_note',
      plannable: makePlannerNote({
        course_id: 999
      })
    });

    const result = transformApiToInternalItem(apiResponse, courses, groups, 'UTC');
    expect(result).toMatchSnapshot();
  });

  it('handles account-level group items', () => {
    const apiResponse = makeApiResponse({
      "context_type": "Group",
      "group_id": "9",
      "plannable_date": "2018-01-12T05:00:00Z",
      "plannable_type": "wiki_page",
      "plannable": makeWikiPage({page_id: "25", html_url:"/groups/9/pages/25"}),
      "html_url": "/groups/9/pages/25"
    });

    const result = transformApiToInternalItem(apiResponse, courses, groups, 'UTC');
    expect(result).toMatchSnapshot();
  });
});

describe('transformInternalToApiItem', () => {
  it('transforms items without a context', () => {
    const internalItem = {
      id: '42',
      date: '2017-05-25',
      title: 'an item',
      details: 'item details',
    };
    const result = transformInternalToApiItem(internalItem);
    expect(result).toMatchObject({
      id: '42',
      todo_date: '2017-05-25',
      title: 'an item',
      details: 'item details',
    });
  });

  it('transforms context information', () => {
    const internalItem = {
      context: {
        id: '42',
      },
    };
    expect(transformInternalToApiItem(internalItem)).toMatchObject({
      context_type: 'Course',
      course_id: '42',
    });
  });
});

describe('transformInternalToApiOverride', () => {
  it('gets override data from an internal item', () => {
    const internalItem = {
      id: '42',
      overrideId: '52',
      type: 'Assignment',
      completed: false
    };
    const result = transformInternalToApiOverride(internalItem, '1');
    expect(result).toMatchObject({
      id: '52',
      plannable_id: '42',
      plannable_type: 'assignment',
      user_id: '1',
      marked_complete: false
    });
  });

  it('graded items should give plannable_id as assignment ID and plannable_type as assignment', () => {
    const internalItem = {
      id: '42',
      overrideId: null,
      type: 'DiscussionTopic',
      overrideAssignId: '10',
      completed: false
    };
    const result = transformInternalToApiOverride(internalItem, '1');
    expect(result).toMatchObject({
      id: null,
      plannable_id: '10',
      plannable_type: 'assignment',
      user_id: '1',
      marked_complete: false
    });
  });

  it('non-graded non-assignment items should give their own ids and types', () => {
    const internalItem = {
      id: '42',
      overrideId: null,
      type: 'Discussion',
      overrideAssignId: null,
      completed: false
    };
    const result = transformInternalToApiOverride(internalItem, '1');
    expect(result).toMatchObject({
      id: null,
      plannable_id: '42',
      plannable_type: 'discussion_topic',
      user_id: '1',
      marked_complete: false
    });
  });
});

describe('transformPlannerNoteApiToInternalItem', () => {
  it('transforms the planner note response to the internal item', () => {
    const apiResponse = makePlannerNoteApiResponse();
    const internalItem = transformPlannerNoteApiToInternalItem(apiResponse, courses, 'UTC');
    expect(internalItem).toMatchSnapshot();
  });

  it('transforms the planner note response to an internal item when the planner note has an associated course', () => {
    const apiResponse = makePlannerNoteApiResponse({ course_id: '1'});
    const internalItem = transformPlannerNoteApiToInternalItem(apiResponse, courses, 'UTC');
    expect(internalItem).toMatchSnapshot();
  });
});
