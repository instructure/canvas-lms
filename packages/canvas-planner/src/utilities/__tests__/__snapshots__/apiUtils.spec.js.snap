// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformApiToInternalItem copes with a non-existent (e.g. concluded) course 1`] = `
Object {
  "completed": false,
  "context": undefined,
  "course_id": 999,
  "date": "2017-05-19T05:59:59Z",
  "dateBucketMoment": "2017-05-19T00:00:00.000Z",
  "details": "Some To Do Note Details :)",
  "html_url": undefined,
  "id": 10,
  "newActivity": false,
  "overrideAssignId": undefined,
  "overrideId": null,
  "points": undefined,
  "status": false,
  "title": "Some To Do Note",
  "toggleAPIPending": false,
  "type": "To Do",
  "uniqueId": "planner_note-10",
}
`;

exports[`transformApiToInternalItem extracts and transforms the proper data for a assignment response 1`] = `
Object {
  "completed": false,
  "context": Object {
    "color": "#abffaa",
    "id": "1",
    "image_url": "blah_url",
    "inform_students_of_overdue_submissions": true,
    "title": "blah",
    "type": "Course",
    "url": undefined,
  },
  "course_id": "1",
  "date": "2017-05-19T05:59:59Z",
  "dateBucketMoment": "2017-05-19T00:00:00.000Z",
  "html_url": "/courses/1/assignments/10",
  "id": "10",
  "newActivity": false,
  "overrideAssignId": undefined,
  "overrideId": null,
  "points": 50,
  "status": false,
  "title": "How to be neutral",
  "toggleAPIPending": false,
  "type": "Assignment",
  "uniqueId": "assignment-10",
}
`;

exports[`transformApiToInternalItem extracts and transforms the proper data for a discussion response 1`] = `
Object {
  "completed": false,
  "context": Object {
    "color": "#abffaa",
    "id": "1",
    "image_url": "blah_url",
    "inform_students_of_overdue_submissions": true,
    "title": "blah",
    "type": "Course",
    "url": undefined,
  },
  "course_id": undefined,
  "date": "2017-05-19T05:59:59Z",
  "dateBucketMoment": "2017-05-19T00:00:00.000Z",
  "html_url": "/courses/1/discussion_topics/10",
  "id": "1",
  "newActivity": false,
  "overrideAssignId": 9,
  "overrideId": null,
  "points": 40,
  "status": false,
  "title": "How to make enemies",
  "toggleAPIPending": false,
  "type": "Discussion",
  "uniqueId": "discussion_topic-1",
  "unread_count": 0,
}
`;

exports[`transformApiToInternalItem extracts and transforms the proper data for a graded discussion reponse with an unread count 1`] = `
Object {
  "completed": true,
  "context": Object {
    "color": "#abffaa",
    "id": "1",
    "image_url": "blah_url",
    "inform_students_of_overdue_submissions": true,
    "title": "blah",
    "type": "Course",
    "url": undefined,
  },
  "course_id": undefined,
  "date": "2017-05-19T05:59:59Z",
  "dateBucketMoment": "2017-05-19T00:00:00.000Z",
  "html_url": "/courses/1/discussion_topics/10",
  "id": "1",
  "newActivity": false,
  "overrideAssignId": 9,
  "overrideId": null,
  "points": 40,
  "status": Object {
    "submitted": true,
    "unread_count": 10,
  },
  "title": "How to make enemies",
  "toggleAPIPending": false,
  "type": "Discussion",
  "uniqueId": "discussion_topic-1",
  "unread_count": 10,
}
`;

exports[`transformApiToInternalItem extracts and transforms the proper data for a graded discussion response 1`] = `
Object {
  "completed": false,
  "context": Object {
    "color": "#abffaa",
    "id": "1",
    "image_url": "blah_url",
    "inform_students_of_overdue_submissions": true,
    "title": "blah",
    "type": "Course",
    "url": undefined,
  },
  "course_id": undefined,
  "date": "2017-05-15T16:32:34Z",
  "dateBucketMoment": "2017-05-15T00:00:00.000Z",
  "html_url": "/courses/1/discussion_topics/10",
  "id": "1",
  "newActivity": false,
  "overrideAssignId": 10,
  "overrideId": null,
  "points": undefined,
  "status": false,
  "title": "How to make friends part 2",
  "toggleAPIPending": false,
  "type": "Discussion",
  "uniqueId": "discussion_topic-1",
  "unread_count": 0,
}
`;

exports[`transformApiToInternalItem extracts and transforms the proper data for a planner_note response 1`] = `
Object {
  "completed": false,
  "context": Object {
    "color": "#abffaa",
    "id": "1",
    "image_url": "blah_url",
    "inform_students_of_overdue_submissions": true,
    "title": "blah",
    "type": "Planner Note",
    "url": undefined,
  },
  "course_id": "1",
  "date": "2017-05-19T05:59:59Z",
  "dateBucketMoment": "2017-05-19T00:00:00.000Z",
  "details": "Some To Do Note Details :)",
  "html_url": undefined,
  "id": 10,
  "newActivity": false,
  "overrideAssignId": undefined,
  "overrideId": null,
  "points": undefined,
  "status": false,
  "title": "Some To Do Note",
  "toggleAPIPending": false,
  "type": "To Do",
  "uniqueId": "planner_note-10",
}
`;

exports[`transformApiToInternalItem extracts and transforms the proper data for a planner_note response without an associated course 1`] = `
Object {
  "completed": false,
  "course_id": undefined,
  "date": "2017-05-19T05:59:59Z",
  "dateBucketMoment": "2017-05-19T00:00:00.000Z",
  "details": "Some To Do Note Details :)",
  "html_url": undefined,
  "id": 10,
  "newActivity": false,
  "overrideAssignId": undefined,
  "overrideId": null,
  "points": undefined,
  "status": false,
  "title": "Some To Do Note",
  "toggleAPIPending": false,
  "type": "To Do",
  "uniqueId": "planner_note-10",
}
`;

exports[`transformApiToInternalItem extracts and transforms the proper data for a quiz response 1`] = `
Object {
  "completed": false,
  "context": Object {
    "color": "#abffaa",
    "id": "1",
    "image_url": "blah_url",
    "inform_students_of_overdue_submissions": true,
    "title": "blah",
    "type": "Course",
    "url": undefined,
  },
  "course_id": "1",
  "date": "2017-05-19T05:59:59Z",
  "dateBucketMoment": "2017-05-19T00:00:00.000Z",
  "html_url": "/courses/1/assignments/10",
  "id": "10",
  "newActivity": false,
  "overrideAssignId": undefined,
  "overrideId": null,
  "points": 100,
  "status": false,
  "title": "How to make friends",
  "toggleAPIPending": false,
  "type": "Quiz",
  "uniqueId": "quiz-10",
}
`;

exports[`transformApiToInternalItem extracts and transforms the proper data for an announcement response 1`] = `
Object {
  "completed": false,
  "context": Object {
    "color": "#abffaa",
    "id": "1",
    "image_url": "blah_url",
    "inform_students_of_overdue_submissions": true,
    "title": "blah",
    "type": "Course",
    "url": undefined,
  },
  "course_id": undefined,
  "date": "2017-05-15T16:32:34Z",
  "dateBucketMoment": "2017-05-15T00:00:00.000Z",
  "html_url": "/courses/1/discussion_topics/10",
  "id": "1",
  "newActivity": false,
  "overrideAssignId": 9,
  "overrideId": null,
  "points": undefined,
  "status": Object {
    "unread_count": 1,
  },
  "title": "",
  "toggleAPIPending": false,
  "type": "Announcement",
  "uniqueId": "announcement-1",
  "unread_count": 1,
}
`;

exports[`transformApiToInternalItem extracts and transforms the proper data for an ungraded discussion reponse with an unread count 1`] = `
Object {
  "completed": false,
  "context": Object {
    "color": "#abffaa",
    "id": "1",
    "image_url": "blah_url",
    "inform_students_of_overdue_submissions": true,
    "title": "blah",
    "type": "Course",
    "url": undefined,
  },
  "course_id": undefined,
  "date": "2017-05-19T05:59:59Z",
  "dateBucketMoment": "2017-05-19T00:00:00.000Z",
  "html_url": "/courses/1/discussion_topics/10",
  "id": "1",
  "newActivity": false,
  "overrideAssignId": 9,
  "overrideId": null,
  "points": undefined,
  "status": Object {
    "unread_count": 10,
  },
  "title": "How to make enemies",
  "toggleAPIPending": false,
  "type": "Discussion",
  "uniqueId": "discussion_topic-1",
  "unread_count": 10,
}
`;

exports[`transformApiToInternalItem handles account-level group items 1`] = `
Object {
  "completed": false,
  "context": Object {
    "color": "#ffeeee",
    "id": "9",
    "image_url": undefined,
    "inform_students_of_overdue_submissions": false,
    "title": "group9",
    "type": "Group",
    "url": "/groups/9",
  },
  "course_id": undefined,
  "date": "2017-06-16 10:08:00Z",
  "dateBucketMoment": "2017-06-16T00:00:00.000Z",
  "html_url": "/groups/9/pages/25",
  "id": "25",
  "newActivity": false,
  "overrideAssignId": undefined,
  "overrideId": null,
  "points": undefined,
  "status": false,
  "title": "wiki_page title",
  "toggleAPIPending": false,
  "type": "Page",
  "uniqueId": "wiki_page-25",
}
`;

exports[`transformPlannerNoteApiToInternalItem transforms the planner note response to an internal item when the planner note has an associated course 1`] = `
Object {
  "completed": false,
  "context": Object {
    "color": "#abffaa",
    "id": "1",
    "image_url": "blah_url",
    "inform_students_of_overdue_submissions": true,
    "title": "blah",
    "type": "Planner Note",
    "url": undefined,
  },
  "course_id": "1",
  "date": "2017-06-21T18:58:51Z",
  "dateBucketMoment": "2017-06-21T00:00:00.000Z",
  "details": "asdfasdfasdf",
  "id": 14,
  "status": false,
  "title": "abc123",
  "type": "To Do",
  "uniqueId": "planner_note-14",
}
`;

exports[`transformPlannerNoteApiToInternalItem transforms the planner note response to the internal item 1`] = `
Object {
  "completed": false,
  "context": Object {},
  "course_id": null,
  "date": "2017-06-21T18:58:51Z",
  "dateBucketMoment": "2017-06-21T00:00:00.000Z",
  "details": "asdfasdfasdf",
  "id": 14,
  "status": false,
  "title": "abc123",
  "type": "To Do",
  "uniqueId": "planner_note-14",
}
`;
